### PR TITLE
Fix issue #24

### LINKED ISSUE #24: Socket in APSEthernet.h binds to first hardware interface
The socket underneath the APSEthernet class [here](https://github.com/BBN-Q/libaps2/blob/8a33ff3a0f9497e1aaa6bcc11f157698afe55c0c/src/lib/APSEthernet.cpp#L3) binds to the first network interface on a system. 

For example on my test laptop eth0 is the wireless interface eth1 is the wired interface. As currently written the socket binds to eth0. flash attempts to send packets out the wireless interface rather than the wired interface (even though the wireless is turned off). 

I have also seen evidence of binding to the first network interface on a windows system.

I changed the endpoint to bind using ip address by using:

```
udp::endpoint(asio::ip::address::from_string("192.168.5.2"))
```

This forced flash and test_comms to used my wired interface.

I recommend that we add a method to set the ip address for binding and a command line switch to programs such as flash and test_comms to allow the user to select the network interface used to send the broadcast packets.

### PR DESCRIPTION
**Pull request by caryan**
Should close #24.  I've tested on Windows 8.1 and Linux Mint 17.1.  As @blakejohnson noted this also provides a way to partition a lab with multiple APS2 setups. 

@blakejohnson could you try on OS X to make sure getifaddrs works the same there.

@bcdonovan could you check it fixes the issues you saw.

### CODE REVIEW

**Comment by blakejohnson** [comment]
As long as I have lightning speed on clicking the OS X firewall exemption button, it works.

**Comment by bcdonovan** [comment]
It solved the problem that I was having under linux.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -27,12 +27,18 @@ INCLUDE_DIRECTORIES("./lib/")
...
 
-#Add flags for gcc and C++11
-ADD_DEFINITIONS(${CMAKE_CXX_FLAGS} "-DGCC")
+#Add flag for C++11
 ADD_DEFINITIONS(${CMAKE_CXX_FLAGS} "-std=c++11")
+
+#mingw64 defaults to WindowsXP SP03 which doesn't support the network calls we want
+#https://github.com/Alexpux/mingw-w64/blob/e960f8fffb399036079cef960a44e86a9abebc71/mingw-w64-headers/include/sdkddkver.h#L145
+#define windows 7 instead
+ADD_DEFINITIONS(${CMAKE_CXX_FLAGS} "-D_WIN32_WINNT=0x0601")
```
Comment (blakejohnson) [inline src/CMakeLists.txt]: should this move inside the `if(WIN32)` block?

File: src/lib/APSEthernet.cpp [hunk 2]
```diff
@@ -182,11 +260,11 @@ int APSEthernet::send_chunk(string serial, vector<APSEthernetPacket> chunk, bool
             seqNum++;
             FILE_LOG(logDEBUG4) << "Packet command: " << print_APSCommand(packet.header.command);
             socket_.send_to(asio::buffer(packet.serialize()), devInfo_[serial].endpoint);
-            // std::this_thread::sleep_for(std::chrono::milliseconds(1));
+            std::this_thread::sleep_for(std::chrono::microseconds(100));
```
Comment (blakejohnson) [inline src/lib/APSEthernet.cpp]: Unrelated change, should be a separate commit if necessary.
